app/src/main/java/com/example/pps_tudai/activity/StepsCounterActivity.java

```diff
@@ -31,28 +36,32 @@ protected void onCreate(Bundle savedInstanceState) {
 
     private void init() {
         ButterKnife.bind(this);
-        StepsCounterView counterView = new StepsCounterView(this);
+        final StepsCounterView counterView = new StepsCounterView(this);
         AppRepository appRepository = new AppRepository(AppRoomDataBase.getDatabase(this).userDao());
         sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
         StepsCounterModel counterModel = new StepsCounterModel(appRepository);
-        presenter = new StepsCounterPresenter(counterView, counterModel, userId, sensorManager);
+        presenter = new StepsCounterPresenter(counterView, counterModel, userId, sensorManager, fusedLocationClient);
     }
 
     @OnClick(R.id.btn_return)
     public void btnReturnClicked() {
         presenter.stopCounter();
+        presenter.stopUpdateLocation();
         presenter.onReturnPressed();
     }
 
     @Override
     protected void onResume() {
         super.onResume();
         presenter.activeCounter();
+        presenter.startLocationUpdates();
     }
 
     @Override
-    protected void onPause() {
-        super.onPause();
+    protected void onStop() {
```

Comment:
instead of make the stop here, please make both stops in a button pls.

Reply:
added a button that allow you to reset the values.